pkg/visibility/api/rest/pending_workloads_lq_test.go

```diff
@@ -0,0 +1,406 @@
+						PositionInClusterQueue: 1,
+						PositionInLocalQueue:   1,
+					}},
+				},
+			},
+		},
+		"valid query parameters set": {
+			clusterQueues: []*kueue.ClusterQueue{
+				utiltesting.MakeClusterQueue(cqNameA).Obj(),
+			},
+			queues: []*kueue.LocalQueue{
+				utiltesting.MakeLocalQueue(lqNameA, nsNameA).ClusterQueue(cqNameA).Obj(),
+			},
+			workloads: []*kueue.Workload{
+				utiltesting.MakeWorkload("a", nsNameA).Queue(lqNameA).Priority(highPrio).Creation(now).Obj(),
+				utiltesting.MakeWorkload("b", nsNameA).Queue(lqNameA).Priority(highPrio).Creation(now.Add(time.Second)).Obj(),
+				utiltesting.MakeWorkload("c", nsNameA).Queue(lqNameA).Priority(highPrio).Creation(now.Add(time.Second * 2)).Obj(),
+			},
+			queryParams: &visibility.PendingWorkloadOptions{
+				Limit: 2,
+			},
+			wantResponse: map[req]*resp{
+				// Only limit is set
+				{
+					nsName:    nsNameA,
+					queueName: lqNameA,
+					queryParams: &visibility.PendingWorkloadOptions{
+						Limit: 2,
+					},
+				}: {wantPendingWorkloads: []visibility.PendingWorkload{
+					{
+						ObjectMeta: v1.ObjectMeta{
+							Name:      "a",
+							Namespace: nsNameA,
+						},
+						LocalQueueName:         lqNameA,
+						Priority:               highPrio,
+						PositionInClusterQueue: 0,
+						PositionInLocalQueue:   0,
+					},
+					{
+						ObjectMeta: v1.ObjectMeta{
+							Name:      "b",
+							Namespace: nsNameA,
+						},
+						LocalQueueName:         lqNameA,
+						Priority:               highPrio,
+						PositionInClusterQueue: 1,
+						PositionInLocalQueue:   1,
+					}},
+				},
+				// Only offset is set
+				{
+					nsName:    nsNameA,
+					queueName: lqNameA,
+					queryParams: &visibility.PendingWorkloadOptions{
+						Offset: 1,
+						Limit:  constants.DefaultPendingWorkloadsLimit,
+					},
+				}: {wantPendingWorkloads: []visibility.PendingWorkload{
+					{
+						ObjectMeta: v1.ObjectMeta{
+							Name:      "b",
+							Namespace: nsNameA,
+						},
+						LocalQueueName:         lqNameA,
+						Priority:               highPrio,
+						PositionInClusterQueue: 1,
+						PositionInLocalQueue:   1,
+					},
+					{
+						ObjectMeta: v1.ObjectMeta{
+							Name:      "c",
+							Namespace: nsNameA,
+						},
+						LocalQueueName:         lqNameA,
+						Priority:               highPrio,
+						PositionInClusterQueue: 2,
+						PositionInLocalQueue:   2,
+					}},
+				},
+				// Both limit and offset are set
+				{
+					nsName:    nsNameA,
+					queueName: lqNameA,
+					queryParams: &visibility.PendingWorkloadOptions{
+						Offset: 1,
+						Limit:  1,
+					},
+				}: {wantPendingWorkloads: []visibility.PendingWorkload{
+					{
+						ObjectMeta: v1.ObjectMeta{
+							Name:      "b",
+							Namespace: nsNameA,
+						},
+						LocalQueueName:         lqNameA,
+						Priority:               highPrio,
+						PositionInClusterQueue: 1,
+						PositionInLocalQueue:   1,
+					}},
+				},
+			},
+		},
+		"invalid queue name": {
```

Comment:
Can we also add a test for invalid query paramters: Limit=xyz, or offset=abc?

Reply:
I see, this is because in this unit test you already have an instance of PendingWorkloadOptions. We could have a unit test for errors in the `conversion_test.go`, but I suggest to do it in follow up.